src/Functions/tokenizeChinese.cpp

```diff
@@ -0,0 +1,103 @@
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypeFixedString.h>
+#include <DataTypes/DataTypeArray.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnFixedString.h>
+#include <Columns/ColumnArray.h>
+#include <Common/Tokenizer/ChineseTokenizer.h>
+#include <Interpreters/Context_fwd.h>
+#include <Interpreters/ITokenExtractor.h>
+#include <Functions/IFunction.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/FunctionFactory.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+}
+
+class FunctionTokenizeChinese : public IFunction
+{
+public:
+    static constexpr auto name = "tokenizeChinese";
+
+    static FunctionPtr create(ContextPtr)
+    {
+        return std::make_shared<FunctionTokenizeChinese>();
+    }
+
+    String getName() const override { return name; }
+
+    size_t getNumberOfArguments() const override { return 1; }
+    bool isVariadic() const override { return false; }
+    ColumnNumbers getArgumentsThatAreAlwaysConstant() const override { return ColumnNumbers{}; }
+
+    bool useDefaultImplementationForNulls() const override { return true; }
+    bool useDefaultImplementationForConstants() const override { return true; }
+    bool useDefaultImplementationForLowCardinalityColumns() const override { return true; }
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
+    {
+        auto input_argument_type = WhichDataType(arguments[0].type);
```

Comment:
For argument validation, let's use `FunctionArgumentDescriptors` from src/Functions/FunctionHelpers.h - it is more descriptive than manual validation. Please see src/Functions/makeDate.cpp for an example how to use it.